### PR TITLE
Enabled Device parsing and improved Browser and O/S parsing

### PR DESCRIPTION
**Pull request by russellwhitaker**
Per specification: https://github.com/ua-parser/uap-core/blob/master/docs/specification.md

Please review @data-wranglers/reviewers thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,16 +67,61 @@ lein run <input_filename> [<optional_out_filename>]
...
 ## Future / Enhancements
 
-What's up next:
-
-1. Implement Device parsing;
-2. Refine Browser and OS parsing to deal with some apparently hairy regex substitution not accounted for in the initial Implementation;
-3. Write a preprocessor for the `speclj` testrunner which filters against text fixtures with no corresponding `regexes.yaml` entry (the need for which will become apparent to the user who runs `lein spec --reporter=d`.)
-
-Pull requests will be very happily considered.
+I have responded to issues posted to this repository with enhancements I rolled into the codebase, so I do welcome feedback from users. Also, pull requests will be very happily considered.
```
Comment (mercul3s) [inline README.md]: Nicely documented. This all makes sense and I was able to follow the steps to clone, pull in dependencies, run test suite, and ran the couple of examples in the repl.

File: spec/uap_clj/core_spec.clj [hunk 2]
```diff
@@ -2,63 +2,207 @@
...
+(def all-tests-browser (:test_cases
                      (parse-string
-                       (slurp ".lein-git-deps/uap-core/tests/test_ua.yaml"))))
+                       (slurp (io/resource "tests/test_ua.yaml")))))
+
+;;;
+;;; These tests are broken in the upsteam uap-core project,
+;;;   and so should not be used in this test suite until they're
+;;;   fixed.
+;;;
```
Comment (mercul3s) [inline spec/uap_clj/core_spec.clj]: Out of curiosity, what in particular is bad about these tests, and what is the scope of work that needs to be done to fix them?

Comment (russellwhitaker) [inline spec/uap_clj/core_spec.clj]: Those tests live upstream in the ua-parser/uap-core project, and would need to be addressed in an issue with the authors like this one I opened - and closed with an accepted PR - earlier: https://github.com/ua-parser/uap-core/issues/51

Basically, these tests don't actually have regexes associated with them that can ever cause these tests to pass. I'll be putting in a PR with the fixes for these, which I expect will be accepted.

Comment (mercul3s) [inline spec/uap_clj/core_spec.clj]: Gotcha.

**Comment by russellwhitaker** [comment]
@mercul3s @data-wranglers/reviewers FYI here's my note to the maintainers of ua-parser/uap-core that they should address the non-specification-compliant presence of 'js_user_agent_string' in a few browser tests that I am filtering against (cf. the latest push where I document that in the test suite): https://github.com/ua-parser/uap-core/issues/55

**Comment by mercul3s** [comment]
LGTM!

**Pull request merged**